manim/manager.py

```diff
@@ -0,0 +1,334 @@
+
+    def _setup(self) -> None:
+        """Set up processes and manager"""
+
+        self.scene.setup()
+
+        # these are used for making sure it feels like the correct
+        # amount of time has passed in the window instead of rendering
+        # at full speed
+        self.virtual_animation_start_time = 0
+        self.real_animation_start_time = time.perf_counter()
+
+    def render(self) -> None:
+        """
+        Entry point to running a Manim class
+
+        Example
+        -------
+
+        .. code-block:: python
+
+            class MyScene(Scene):
+                def construct(self):
+                    self.play(Create(Circle()))
+
+
+            with tempconfig({"preview": True}):
+                Manager(MyScene).render()
+        """
+        self._render_first_pass()
+        self._render_second_pass()
+        self._interact()
+
+    def _render_first_pass(self) -> None:
+        """
+        Temporarily use the normal single pass
+        rendering system
+        """
+        self._setup()
+
+        try:
+            self.scene.construct()
+            self._post_contruct()
+            self._interact()
+        except EndSceneEarlyException:
+            pass
+        self._tear_down()
+
+    def _render_second_pass(self) -> None:
+        """
+        In the future, this method could be used
+        for two pass rendering
+        """
+        ...
+
+    def _post_contruct(self) -> None:
+        self.file_writer.finish()
+        self._write_files = False
+
+    def _tear_down(self) -> None:
+        self.scene.tear_down()
+
+        if config.save_last_frame:
+            self._update_frame(0)
+
+        if self.window is not None:
+            self.window.close()
+            self.window = None
+
+    def _interact(self) -> None:
+        if self.window is None:
+            return
+        logger.info(
+            "\nTips: Using the keys `d`, `f`, or `z` "
+            + "you can interact with the scene. "
+            + "Press `command + q` or `esc` to quit"
+        )
+        self.scene.skip_animations = False
+        self.scene.refresh_static_mobjects()
+        while not self.window.is_closing:
+            # TODO: Replace with actual dt instead
+            # of hardcoded dt
+            dt = 1 / config.frame_rate
+            self._update_frame(dt)
+
+    def _update_frame(self, dt: float, *, write_to_file: bool | None = None) -> None:
+        """Update the current frame by ``dt``
+
+        Parameters
+        ----------
+            dt : the time in between frames
+            write_to_file : Whether to write the result to the output stream.
+                Default value checks :attr:`_write_files` to see if it should be written.
+        """
+        self.time += dt
+        self.scene._update_mobjects(dt)
+
+        if self.window is not None:
+            self.window.clear()
+
+        state = self.scene.get_state()
+        self._render_frame(state, write_file=write_to_file)
```

Comment:
Instead of another parameter, this should just check `if dt` so that it renders faster